pkg/operator/externaletcdsupportcontroller/external_etcd_support_controller.go

```diff
@@ -0,0 +1,192 @@
+
+	c := &ExternalEtcdEnablerController{
+		operatorClient:        operatorClient,
+		targetImagePullSpec:   targetImagePullSpec,
+		operatorImagePullSpec: operatorImagePullSpec,
+		envVarGetter:          envVarGetter,
+		kubeClient:            kubeClient,
+	}
+	syncCtx := factory.NewSyncContext("ExternalEtcdSupportController", eventRecorder.WithComponentSuffix("external-etcd-support-controller"))
+	c.enqueueFn = func() {
+		syncCtx.Queue().Add(syncCtx.QueueKey())
+	}
+	envVarGetter.AddListener(c)
+	syncer := health.NewDefaultCheckingSyncWrapper(c.sync)
+
+	return factory.New().
+		WithSyncContext(syncCtx).
+		ResyncEvery(time.Minute).
+		WithSync(syncer.Sync).
+		WithInformers(
+			operatorClient.Informer(),
+			kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints().Informer(),
+			kubeInformersForOpenshiftEtcdNamespace.Core().V1().ConfigMaps().Informer(),
+			kubeInformersForOpenshiftEtcdNamespace.Core().V1().Secrets().Informer(),
+			masterNodeInformer,
+			infrastructureInformer.Informer(),
+			networkInformer.Informer(),
+		).ToController("ExternalEtcdController", eventRecorder.WithComponentSuffix("external-etcd-controller"))
+}
+
+func (c *ExternalEtcdEnablerController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+
+	operatorSpec, _, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	envVars := c.envVarGetter.GetEnvVars()
+	if len(envVars) == 0 {
+		// note this will not degrade the controller, that can happen during CEO restarts often due to cold informer caches (expected)
+		return fmt.Errorf("ExternalEtcdEnablerController missing env var values")
+	}
+
+	podSub := c.getPodSubstitution(c.targetImagePullSpec, c.operatorImagePullSpec, envVars)
+	_, _, err = c.supportExternalPod(ctx, podSub, c.kubeClient.CoreV1(), syncCtx.Recorder(), operatorSpec)
+	if err != nil {
+		err = fmt.Errorf("%q: %w", "configmap/external-etcd-pod", err)
+	}
+
+	return err
+}
+
+func (c *ExternalEtcdEnablerController) getPodSubstitution(
+	imagePullSpec, operatorImagePullSpec string,
+	envVarMap map[string]string) *targetconfigcontroller.PodSubstitutionTemplate {
+
+	var nameValues []targetconfigcontroller.NameValue
+	for _, k := range sets.StringKeySet(envVarMap).List() {
+		v := envVarMap[k]
+		nameValues = append(nameValues, targetconfigcontroller.NameValue{k, v})
+	}
+
+	return &targetconfigcontroller.PodSubstitutionTemplate{
+		Image:               imagePullSpec,
+		OperatorImage:       operatorImagePullSpec,
+		ListenAddress:       "0.0.0.0",   // TODO this needs updating to detect ipv6-ness
+		LocalhostAddress:    "127.0.0.1", // TODO this needs updating to detect ipv6-ness
+		LogLevel:            "info",
+		EnvVars:             nameValues,
+		EnableEtcdContainer: true,
+	}
+}
+
+func (c *ExternalEtcdEnablerController) supportExternalPod(ctx context.Context, subs *targetconfigcontroller.PodSubstitutionTemplate,
```

Comment:
I wonder why we need to replicate all the logic here, can't you copy the rendered pod template that we "ignore" from in `manageStandardPod` of the targetconfigcontroller?

Reply:
Moreover this new function render the data in JSON format, not in YAML, as the Node does not have jq and an external etcd manager (like the pacemaker agent in TNF) needs to edit the manifest

Reply:
I leave this to Allen, but I personally prefer to have the templating logic not duplicated across two concurrently running controllers. We can serialize json or yaml on the other controller too.

Reply:
I could rework both functions to collect same/similar code in a third function and reuse that.
Maybe ceohelpers.go is a good fit?

Reply:
I moved the common logic into a new file under ceohelpers `pkg/operators/ceohelpers/podsubstitution.go`